### ISSUE TITLE
[BUG] TerminalCloudAPI: Unable to pass SaleToAcquirerData metadata with request.

### ISSUE THREAD

**Issue by sarachai** [Adyen/adyen-node-api-library#703]
**Describe the bug**
Unable to pass in `metadata` with the `saleToAcquirerData` field (as per: https://docs.adyen.com/point-of-sale/add-data/sale-to-acquirer-data).

**To Reproduce**
Create a TerminalCloudAPI sync or async request with the following `saleData`:
```javascript
            saleToAcquirerData: {
              metadata: {
                someMetaDataKey1: `pikachu`,
                someMetaDataKey2: `incredible`,
              }
            }
```

**Expected behavior**
The metadata is present within the `SaleToPOIResponse` (if using the sync endpoint), and/or the Adyen webhook notification `additionalData` object.

**Additional context**
I noticed the metadata field is not included within the `SaleToAcquirerData` typing.

Thank you!!!

**Comment by wboereboom** [comment]
HI @sarachai , Thanks for raising this issue. 
I checked and the types are indeed missing some fields. I shall regenerate all types for the TerminalCloudAPI and make sure the metadata object is included.

Regards,
Wouter
Adyen

**Comment by wboereboom** [comment]
It seems there is a big discrepancy between the types generated from the API specs and the types in the library. I'm updating only the type definition that causes issues for now, and will create a task to look into type generation for this API service in the future.

**Comment by wboereboom** [comment]
The issue should be fixed in release v8.1.1.

Regards,
Wouter
Adyen

**Comment by sarachai** [comment]
It works perfectly now — thank you!!
